File: src/main/scala/org/openapitools/client/api/VisitList.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import VisitList._

case class VisitList (
  metadata: Option[Metadata],
pagination: Option[Pagination],
data: Option[List[Visit]])

object VisitList {
  import DateTimeCodecs._

  implicit val VisitListCodecJson: CodecJson[VisitList] = CodecJson.derive[VisitList]
  implicit val VisitListDecoder: EntityDecoder[VisitList] = jsonOf[VisitList]
  implicit val VisitListEncoder: EntityEncoder[VisitList] = jsonEncoderOf[VisitList]
}

File: client/javascript-closure-angular/API/Client/VisitList.js
goog.provide('API.Client.VisitList');

/**
 * @record
 */
API.Client.VisitList = function() {}

/**
 * @type {!API.Client.Metadata}
 * @export
 */
API.Client.VisitList.prototype.metadata;

/**
 * @type {!API.Client.Pagination}
 * @export
 */
API.Client.VisitList.prototype.pagination;

/**
 * @type {!Array<!API.Client.Visit>}
 * @export
 */
API.Client.VisitList.prototype.data;

